doc-site/docs/tutorials/private-storage.md

```diff
@@ -0,0 +1,180 @@
+```
+
+#### Key Points  
+
+1. The contract is deployed **inside the privacy group**, meaning **only group members** can interact with it.  
+2. **Transactions involving this contract are private** and only visible to **Node1 & Node2**.  
+
+---
+
+## Step 3: Store and Retrieve Values as Group Members
+
+### Storing a Value
+
+Now that the contract is deployed, **Node1** can store a value.
+
+```typescript
+const privateStorage = new PrivateStorage(memberPrivacyGroup, contractAddress);
+
+const valueToStore = 125; // Example value to store
+logger.log(`Storing a value "${valueToStore}" in the contract...`);
+const storeTx = await privateStorageContract.sendTransaction({
+  from: verifierNode1.lookup,
+  function: "store",
+  data: { num: valueToStore }
+});
+logger.log("Value stored successfully! Transaction hash:", storeTx?.transactionHash);
+```
+
+---
+
+### Retrieving the Stored Value
+
+Group members **Node1 & Node2** can now retrieve the stored value.
+
+```typescript
+// Retrieve the value as Node1
+logger.log("Node1 retrieving the value from the contract...");
+const retrievedValueNode1 = await privateStorageContract.call({
+    from: verifierNode1.lookup,
+    function: "retrieve"
+  }
+);
+logger.log("Node1 retrieved the value successfully:", retrievedValueNode1["value"]);
+
+// Retrieve the value as Node2
+logger.log("Node2 retrieving the value from the contract...");
+const retrievedValueNode2 = await privateStorageContract
+  .using(paladinNode2)
+  .call({
+    from: verifierNode2.lookup,
+    function: "retrieve",
+  });
+logger.log("Node2 retrieved the value successfully:", retrievedValueNode2["value"]);
+```
+
+---
+
+## Step 4: Verify Privacy by Testing Unauthorized Access
+
+Now, let’s test if **Node3 (an outsider)** can access the stored data.  
+
+### **What should happen?**  
+
+Node3 should NOT be able to retrieve the stored value because it wasn’t part of the privacy group.  
+
+```typescript
+try {
+  logger.log("Node3 (outsider) attempting to retrieve the value...");
+     await privateStorageContract.using(paladinNode3).call({
+      from: verifierNode3.lookup,
+      function: "retrieve",
+  });
+  logger.error("Node3 (outsider) should not have access to the private Storage contract!");
+  return false;
+} catch (error) {
+  logger.info("Expected error - Node3 (outsider) cannot retrieve the data. Access denied.");
+}
+```
+
+### Why Privacy Groups Work
+- **Private State Isolation** – Transactions within a privacy group are only visible to its members.  
+- **No Global State Sharing** – Outsiders (e.g., Node3) never receive the transaction history, making it impossible for them to infer contract data.  
+- **Selective State Distribution** – Only group members can access and verify the shared state.  
+
+By design, **Node3 does not just “lack permission” to call the contract—it lacks any knowledge of its state, history, or data, making unauthorized access fundamentally impossible**.  
```

Comment:
We may need to iterate on how we explain privacy groups. This is good for now, but I do think it's probably the most misunderstood part of Paladin right now. The fact that they're not quite blockchains, not quite rollups, not _really_ "deployed" in the sense you normally think of it, but they do have a state tree that is shared privately and progressed forward confidentially between the members... maybe this isn't the place to get into all of that, but we need to make sure we eventually point people to that level of detail.

Reply:
I agree. I believe we'll need to revisit and refine this based on how the industry responds to this and similar solutions as they evolve. As the public gains a better understanding of the mechanisms involved, it will become clearer. It’s challenging to convey new concepts to the industry in just a few words within a tutorial.